File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length==1) return strs[0];
        String result = strs[0];
        for(int i=1; i<strs.length; i++){
            result = findCommonPrefix(result,strs[i]);
        }
        return result;
    }

    public String findCommonPrefix(String str1, String str2){
        int minLength = str1.length()<str2.length() ? str1.length() : str2.length();
        for(int i=0; i<minLength; i++){
            if(str1.charAt(i)!=str2.charAt(i)){
                return str1.substring(0,i);
            }
        }
        return str1.substring(0,minLength);
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        n=len(strs)
        lcp=strs[0]
        for i in range(1,n):
            j=0
            while j<len(lcp) and j<len(strs[i]):
                if lcp[j]!=strs[i][j]:
                    break
                j+=1
            lcp=lcp[0:j]
        return lcp